chat-android/src/test/java/com/ably/chat/Sandbox.kt

```diff
@@ -0,0 +1,45 @@
+package com.ably.chat
+
+import com.google.gson.JsonElement
+import com.google.gson.JsonParser
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.request.delete
+import io.ktor.client.request.get
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.client.statement.HttpResponse
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+
+val client = HttpClient(CIO)
+
+class Sandbox {
+    private lateinit var appId: String
+
+    lateinit var apiKey: String
+        private set
+
+    suspend fun setUp() {
```

Comment:
I feel constructor for `Sandbox` class should be private and sandbox should only be accessed when it's properly setup. 
Sandbox should be initialized and returned as a part of static method . i.e. `companion object`
```suggestion
    suspend fun Sandbox setUp() {
```

Reply:
Btw, I would like to highlight that, creating the sandbox sometimes fails (it's common). We have faced similar issues in the past leading to flaky integration test behavior in the case of ably-go. We can implement a maxRetryCount=4 similar to the [ably-go sandbox retry count](https://github.com/ably/ably-go/blob/a81632dfeb26bd545232674622463e5709d42e03/ablytest/sandbox.go#L152), while catching the current `setup` code block as a Throwable.

Reply:
added retries